main.go

```diff
@@ -218,6 +224,27 @@ func createProxyConfig(options Options) proxy.Config {
 		}
 	}
 
+	if len(options.EDNSOpt) > 0 {
+		config.EDNSOpts = make(map[uint16][]byte)
```

Comment:
Could you please move this code to a separate function and also add a unit-test for it?